domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+   *   assigning its users to the default domain.
+   *
+   * @usage drush domain:delete --content-as=example_net --users-as=example_net
+   *   example.com Delete the domain example.com, assigning its content and
+   *   users to the example.net domain.
+   *
+   * @usage drush domain:delete --dryrun 19476
+   *   Show the effects of delete the domain example.com and assigning its
+   *   content and users to the default domain, example.org, but not doing so.
+   *
+   * @usage drush domain:delete --chatty example_net
+   *   Verbosely Delete the domain example.net and assign its content and users
+   *   to the default domain, example.org.
+   *
+   * @usage drush domain-delete --chatty all
+   *   Verbosely Delete the domains example.com and example.net and assign
+   *   their content and users to the default domain, example.org.
+   *
+   * @option chatty
+   *   Document each step as it is performed.
+   * @option dryrun
+   *   Do not do anything, but explain what would be done. Implies --chatty.
+   * @option content-as
+   *   Values "prompt", "ignore", "default", <name>, Reassign content
+   *   associated with the the domain being deleted to the default domain, to
+   *   a specified domain, or leave the content alone (&hence inaccessible in
+   *   the normal way). The default value is 'prompt': ask which domain to use.
+   * @option users-as
+   *   Values "prompt", "ignore", "default", <name>, Reassign user accounts
+   *   associated with the the domain being deleted to the default domain,
+   *   to the domain whose machine name is <name>, or leave the user accounts
+   *   alone (&hence inaccessible in the normal way). The default value is
+   *   'prompt': ask which domain to use.
+   *
+   * @command domain:delete
+   * @aliases domain-delete
+   * 
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function delete($domain_id, $options = ['content-as' => null, 'users-as' => null, 'dryrun' => null, 'chatty' => null ]) {
+    $policy_content = 'prompt';
+    $policy_users = 'prompt';
+
+    $this->is_dry_run = (bool)$options['dryrun'];
+
+    //region Get current domain list and perform sanity checks.
+    $domain_storage = $this->getStorage();
+    $default_domain = $domain_storage->loadDefaultDomain();
+    $all_domains = $domain_storage->loadMultipleSorted(NULL);
+
+    if (empty($all_domains)) {
+      throw new DomainCommandException('There are no configured domains.');
+    }
+    if (empty($domain_id)) {
+      throw new DomainCommandException('You must specify a domain to delete.');
+    }
+    //endregion
+
+    //region Determine which domains to be deleted.
+    if ($domain_id == 'all') {
+      $domains = $all_domains;
+      if (empty($domains)) {
+        $this->logger()->info(dt('There are no domains to delete.'));
+        return;
+      }
+      $really = $this->io()->confirm(dt('This action cannot be undone. Continue?:'), FALSE);
```

Comment:
What is `$this->io()`?

Reply:
It is part of the new drush 9 command infrastructure, which replaces the old drush_print, drush_choice et al. The call returns a configured IO object which is a subclass of SymfonyStyle, and is provided by the DrushCommands class.